src/middleware/rateLimiterSetup.ts

```diff
@@ -57,6 +146,17 @@ export default function setupRateLimiter(
                 // typescript should never let us invoke this function with anything other than the options above
                 throw new Error('Selected rate limiting algorithm is not suppported');
         }
+
+        const boundProcessRequest = rateLimiter.processRequest.bind(rateLimiter);
+
+        rateLimiter.processRequest = async (
```

Comment:
Could you make a note that we are overriding the previous `processRequest` method with this throttled implementation and not adding a totally new property to the rate-limiter here.